File: 프로그래머스/구명보트.py
def solution(people, limit):
    answer = 0
    
    people.sort()
    
    minidx = 0
    maxidx = len(people)-1
    
    while minidx <= maxidx:
        
        if people[maxidx] + people[minidx] <= limit:
            minidx += 1
        
        answer+=1
        maxidx-=1
            
    return answer

File: 프로그래머스/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        
        Arrays.sort(people);
        
        int minidx = 0;
        
        for(int maxidx = people.length-1; minidx <= maxidx; maxidx--){
            if(people[maxidx] + people[minidx] <= limit)
                minidx++;
            answer++;
        }
        return answer;
    }
}